src/EscapeHit.Invoice.Database/Configuration/InvoiceEntityConfiguration.cs

```diff
@@ -2,16 +2,22 @@
 using Microsoft.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore.Metadata.Builders;
 using AppForeach.Framework.EntityFrameworkCore.DataType;
+using EscapeHit.Invoice.Specification;
 
 namespace EscapeHit.Invoice.Database.Configuration
 {
     public class InvoiceEntityConfiguration : IEntityTypeConfiguration<InvoiceEntity>
     {
-        public void Configure(EntityTypeBuilder<InvoiceEntity> builder)
+        private readonly InvoiceEntitySpecification _entitySpecification;
+
+        public InvoiceEntityConfiguration(InvoiceEntitySpecification entitySpecification)
         {
-            //materialze FacetBag
+            _entitySpecification = entitySpecification;
```

Comment:
pass entity specification from DI

Reply:
we need to investigate if we can get rid of this constructor and passing specification in it, especially as we surely would like to retain support for modelBuilder.ApplyConfigurationsFromAssembly

Reply:
At the moment EF constructor receives IServiceLocator, from which InvoiceEntitySpecification is being retrieved, but I haven't tested this ApplyConfigurationsFromAssembly approach yet.
![image](https://github.com/user-attachments/assets/3b7b141c-c979-41ce-9807-383e077e7407)